src/drawers/drawer.ts

```diff
@@ -77,12 +81,11 @@ export module Drawers {
       return data.length;
     }
 
-    private _applyMetadata(attrToProjector: AttributeToProjector,
-                          dataset: Dataset): AttributeToAppliedProjector {
+    private _appliedProjectors(attrToProjector: AttributeToProjector): AttributeToAppliedProjector {
       var modifiedAttrToProjector: AttributeToAppliedProjector = {};
       d3.keys(attrToProjector).forEach((attr: string) => {
         modifiedAttrToProjector[attr] =
-          (datum: any, index: number) => attrToProjector[attr](datum, index, dataset);
+          (datum: any, index: number) => attrToProjector[attr](datum, index, this.__dataset);
```

Comment:
This is inside the class where the instance variable is, and we're currently undecided on if we want to stick with getters or setters.

That said, I'm ok with using a getter here and I think using getters is more consistent even though it's slightly less performant